OOP/HW_1.py

```diff
@@ -0,0 +1,546 @@
+# 
+# Подсказка: если в процессе гуглёжки, вы выйдете на такую тему как **"Дескрипторы", то это НЕ то, что вам сейчас нужно**
+# 
+# Примеры использования ниже
+
+# In[17]:
+
+
+class StrangeFloat(float):
+    def __getattribute__(self, name):
+        if hasattr(float, name):
+            return super().__getattribute__(name)
+        else:
+            operation, number = name.split('_')
+            number = float(number)
+            operation_dict = {
+                'add': self.__add__,
+                'subtract': self.__sub__, 
+                'multiply': self.__mul__,
+                'divide': self.__truediv__
+            }
+            return StrangeFloat(operation_dict[operation](number))
+            
+
+
+# In[18]:
+
+
+number = StrangeFloat(3.5)
+
+
+# In[19]:
+
+
+number.add_1
+
+
+# In[20]:
+
+
+number.subtract_20
+
+
+# In[21]:
+
+
+number.multiply_5
+
+
+# In[22]:
+
+
+number.divide_25
+
+
+# In[23]:
+
+
+number.add_1.add_2.multiply_6.divide_8.subtract_9
+
+
+# In[24]:
+
+
+getattr(number, "add_-2.5")   # Используем getattr, так как не можем написать number.add_-2.5 - это SyntaxError
+
+
+# In[25]:
+
+
+number + 8   # Стандартные для float операции работают также
+
+
+# In[26]:
+
+
+number.as_integer_ratio()   # Стандартные для float операции работают также  (это встроенный метод float, писать его НЕ НАДО)
+
+
+# # Задание 4 (3 балла)
+
+# В данном задании мы немного отдохнём и повеселимся. От вас требуется заменить в данном коде максимально возможное количество синтаксических конструкций на вызовы dunder методов, dunder атрибутов и dunder переменных.
+# 
+# Маленькая заметка: полностью всё заменить невозможно. Например, `function()` можно записать как `function.__call__()`, но при этом мы всё ещё не избавляемся от скобочек, так что можно делать так до бесконечности `function.__call__.__call__.__call__.__call__.....__call__()` и при всём при этом мы ещё не избавляемся от `.` для доступа к атрибутам. В общем, замените всё, что получится, не закапываясь в повторы, как в приведённом примере. Чем больше разных методов вы найдёте и используете, тем лучше и тем выше будет балл
+# 
+# Код по итогу дожен работать и печатать число **4420.0**, как в примере. Структуру кода менять нельзя, просто изменяем конструкции на синонимичные
+# 
+# И ещё маленькая подсказка. Заменить здесь можно всё кроме:
+# + Конструкции `for ... in ...`:
+# + Синтаксиса создания лямбда функции
+# + Оператора присваивания `=`
+# + Конструкции `if-else`
+
+# In[35]:
+
+
+np = __import__('numpy')
+
+
+matrix = []
+for idx in range(0, 100, 10):
+    matrix.__iadd__([list(range(idx, idx.__add__(10)))])
+    
+selected_columns_indices = list(filter(lambda x: x in range(1, 5, 2), range(matrix.__len__())))
+selected_columns = map(lambda x: [x.__getitem__(col) for col in selected_columns_indices], matrix)
+
+arr = np.array(list(selected_columns))
+
+mask = (np.array(list(i.__getitem__(1) for i in arr)).__mod__(3)).__eq__(0)
```

Comment:
Тут ты изменила суть, здесь не делается срез, а создаётся новый эррей

```suggestion
mask = arr.__getitem__((slice(None, None, None), 1)).__mod__(3)).__eq__(0)
```